sympy/functions/combinatorial/factorials.py

```diff
@@ -541,6 +561,8 @@ def eval(cls, x, k):
 
         if x is S.NaN:
             return S.NaN
+        elif x == k:
+            return factorial(x)
         elif k.is_Integer:
             if k is S.NaN:
```

Comment:
That's not possible. I will go through ``eval`` method. This should have been caught. Probably this is not very well tested.